src/utils/cloud_storage.R

```diff
@@ -0,0 +1,181 @@
+box::use(googlesheets4)
+box::use(googledrive)
+box::use(dplyr)
+box::use(stringr)
+box::use(readr)
+box::use(arrow)
+box::use(gcs = googleCloudStorageR)
+
+box::use(../utils/gmas_test_run[gmas_test_run])
+
+#' Read a Parquet file stored on Google Drive
+#'
+#' Reads a file from the `global-monitoring` bucket.
+#' The file is read based on its prefix. Currently, the only support is for
+#' Apache Parquet files, but other support can be added if necessary.
+#'
+#' Function parsing is done based on file type:
+#'
+#' * Apache Parquet: [arrow::read_parquet()].
+#'
+#' @param name Name of the file to read, including prefix (`input/` or `output/`)
+#'     and filetype `.parquet`.
+#'
+#' @returns Data frame.
+#'
+#' @export
+read_gcs_file <- function(name) {
+  gcs$gcs_get_object(
+    object_name = name,
+    bucket = "global-monitoring",
+    parseFunction = gcs_parse_parquet
+  )
+}
+
+#' Parse parquet file
+#'
+#' Helper function to parse parquet file.
+#'
+#' @param object Object to parse
+gcs_parse_parquet <- function(object) {
+  con <- gzcon(rawConnection(httr::content(object)))
```

Comment:
never seen this! is it required to read in parquet from GCS?

Reply:
Yeah, you could either separately download the file without any parsing, and then just read in. Or you can specifically create a parsing function that you just pass in to the function that reads from the connection.

Reply:
I wonder if we can use the approach outlined here which allow  you to read the parquet files in directly?

https://arrow.apache.org/docs/10.0/r/articles/fs.html

I couldn't get it to work in the global-monitoring bucket, but i am wondering if that has something to do with permissions settings? because the examples in the article do work